src/test/qgstest.h

```diff
@@ -292,6 +296,54 @@ class TEST_EXPORT QgsTest : public QObject
       return result;
     }
 
+    bool checkLongStr( const char *file, const char *function, int line, const QString &name, const QString &referenceName, const QByteArray &actualStr )
+    {
+      QString subPath = "control_images/" + mControlPathPrefix + "/expected_" + name + "/" + "expected_" + referenceName;
```

Comment:
Shouldn't this be a different (non "image") path here?

Reply:
It might but this will increase the number of directories in `testdata`. I told myself it would be better to have all expected content in the same place when you already have a test that checks against images. 

I found the `testdata` directory a bit messy and I did not want to add a new directory inside.  

What do you suggest?

Reply:
> I found the testdata directory a bit messy

No argument there :laughing: 

> What do you suggest?

I'd still lean toward a new "reference_files" or "control_files" subfolder. I think putting non-images in the control_images folder will just make the mess worse

Reply:
Lets go for a "control_files" subfolder!